my-project/src/Controller/ProductCategoryCSVController.php

```diff
@@ -71,4 +72,50 @@ public function exportCSV(ProductCategory $productCategory, SerializerService $s
             $dataJson
             ,200);
     }
+
+    /**
+     * @Route("/", name="product_category_import_csv_api", methods="POST")
+     */
+    public function importFromCSV(Request $request, SerializerInterface $serializer, ValidatorInterface $validator)
+    {
+        $productCategoriesArray = $serializer->decode(file_get_contents($request->get('csvFile')), 'csv');
+        $em = $this->getDoctrine()->getManager();
+        $tmpProductsCategories = [];
+        foreach ($productCategoriesArray as $productCategory) {
+            $tmpProductCategory = new ProductCategory();
```

Comment:
it could be done in a for loop with general method setData() than i wouldnt need modification when fields change, doesnt have to be done, just something to consider